tools/probes/probes_main.c

```diff
@@ -0,0 +1,306 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2019 Intel Corporation. All rights reserved.
+//
+// Author: Adrian Bonislawski <adrian.bonislawski@intel.com>
+
+/*
+ * Probes will extract data for several probe points in one stream
+ * with extra headers. This app will read the resulting file,
+ * strip the headers and create wave files for each extracted buffer.
+ *
+ * Usage to parse data and create wave files: ./sof-probes -p data.bin
+ *
+ */
+
+#include <ipc/probe.h>
+#include <sof/math/numbers.h>
+#include "wave.h"
+
+#include <ctype.h>
+#include <errno.h>
+#include <fcntl.h>
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdio.h>
+#include <stdint.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+
+#define APP_NAME "sof-probes"
+
+#define PACKET_MAX_SIZE	4096	/**< Size limit for probe data packet */
+#define DATA_READ_LIMIT 1024	/**< Data limit for file read */
+#define FILES_LIMIT	8	/**< Maximum num of files */
```

Comment:
shouldn't this be aligned with the max number of extractors?

Reply:
yes, it should but it is generated by kconfig in build dir which is unknown for this app. Maybe I will just increase this limit, ok?